contracts/serverless-v2/Executors.sol

```diff
@@ -462,19 +458,19 @@ contract Executors is
         address _jobOwner
     ) internal {
         if(!_hasExecutedJob) {
+            uint256 totalComp = executors[_executor].stakeAmount * SLASH_PERCENT_IN_BIPS / SLASH_MAX_BIPS;
+            executors[_executor].stakeAmount -= totalComp;
+
             if(_isNoOutputSubmitted) {
-                executors[_executor].stakeAmount -= (SLASH_COMP_FOR_JOB_OWNER + SLASH_COMP_FOR_GATEWAY);
-                
-                // transfer the slashed comp to job owner
-                TOKEN.safeTransfer(_jobOwner, SLASH_COMP_FOR_JOB_OWNER);
                 // transfer the slashed comp to gateway that relayed the job
                 TOKEN.safeTransfer(_gateway, SLASH_COMP_FOR_GATEWAY);
+                // transfer the slashed comp to job owner
+                TOKEN.safeTransfer(_jobOwner, totalComp - SLASH_COMP_FOR_GATEWAY);
+            }
+            else {
+                // transfer the slashed comp to common pool(jobs contract)
+                TOKEN.safeTransfer(_msgSender(), totalComp);
             }
-
-            uint256 commonPoolComp = executors[_executor].stakeAmount * SLASH_PERCENT_IN_BIPS / SLASH_MAX_BIPS;
-            executors[_executor].stakeAmount -= commonPoolComp;
-            // transfer the slashed comp to common pool(jobs contract)
-            TOKEN.safeTransfer(_msgSender(), commonPoolComp);
         }
 
         _releaseExecutor(_executor);
```

Comment:
_releaseExecutor should go inside the if block. Because, for the executor which has already submitted the output, it is incorrect to call releaseExecutor on it.

Better, if you can call slashExecutor selectively from jobs contract, only for executors who haven't submitted the output.

Reply:
hmm, seems I made a big mess here😬. Updated the logic.